### PR TITLE
add text classification analyzer

### LINKED ISSUE #77: [Analyzer] Integrate intent classifier
An intent classifier is useful to detect what customer goals like buy, sell, and purchase also useful in conversional flow. 

This same things can also be done via zero shot classifier as well but it would better to add separate analyzer to separate it from generic text classification. It will help user to load their own models for this purpose.

### PR DESCRIPTION
**Pull request by shahrukhx01**
The initial draft for text classifier which will resolve #77

### CODE REVIEW

File: obsei/analyzer/classification_analyzer.py [hunk 1]
```diff
@@ -121,7 +122,106 @@ def analyze_input(  # type: ignore[override]
...
+
+        analyzer_output: List[TextPayload] = []
+
+        labels = analyzer_config.labels or []
+        add_positive_negative_labels = kwargs.get("add_positive_negative_labels", True)
+        if add_positive_negative_labels:
+            if "positive" not in labels:
+                labels.append("positive")
+            if "negative" not in labels:
+                labels.append("negative")
```
Comment (lalitpagaria) [inline obsei/analyzer/classification_analyzer.py:177]: Not needed here

**Comment by shahrukhx01** [comment]
@lalitpagaria PR is complete from my side, one possible improvement that can be added. is mapping of labels, transformers map labels as {"LABEL_0", "LABEL_1",....,"LABEL_N"}, what we can do is use the labels parameter and map values from labels to transformers labels, however, if the user passes the wrong sequence of labels, they'd get altogether wrong predictions. So there must be some mechanism for enforcing the right sequence in my knowledge it won't be possible to enforce such a check. what do you think?

**Comment by lalitpagaria** [comment]
@shahrukhx01 I have refactored you changes a little. Also added functionality to support user provided label map. Check test and let me know if you have any concern. Post that I will merge this.
Also it would be great if we plan a demo/article/colab showing capabilities of https://huggingface.co/obsei-ai/sell-buy-intent-classifier-bert-mini

Or we can show demo of newly added classifier to detect hate speech via https://huggingface.co/Hate-speech-CNERG/dehatebert-mono-english
So workflow would be Twitter or Reddit -> Analyzer -> Elastic Search or Slack

**Review by lalitpagaria — APPROVED** [review]
LGTM!

**Comment by shahrukhx01** [comment]
@lalitpagaria I can write a small colab tutorial and the post it on LinkedIn about buy-sell intent model.

**Comment by lalitpagaria** [comment]
@shahrukhx01 We can add tutorial here https://github.com/obsei/obsei#tutorials and article here https://github.com/obsei/obsei#articles

**Pull request merged**